thrust/thrust/device_reference.h

```diff
@@ -947,17 +947,16 @@ class device_reference : public thrust::reference<T, thrust::device_ptr<T>, thru
      */
     device_reference &operator^=(const T &rhs);
 #endif // end doxygen-only members
-}; // end device_reference
 
-/*! swaps the value of one \p device_reference with another.
- *  \p x The first \p device_reference of interest.
- *  \p y The second \p device_reference of interest.
- */
-template <typename T>
-_CCCL_HOST_DEVICE void swap(device_reference<T>& x, device_reference<T>& y)
-{
-  x.swap(y);
-}
+  /*! swaps the value of one \p device_reference with another.
+   *  \p x The first \p device_reference of interest.
+   *  \p y The second \p device_reference of interest.
+   */
+  friend _CCCL_HOST_DEVICE void swap(device_reference& x, device_reference& y) noexcept(noexcept(x.swap(y)))
```

Comment:
Nitpick, we generally move attributes to the front
```suggestion
  _CCCL_HOST_DEVICE friend void swap(device_reference& x, device_reference& y) noexcept(noexcept(x.swap(y)))
```